### PR TITLE
Merge ModuleCatalog into ComponentCatalog.

### PR DESCRIPTION
**Pull request by eerhardt**
This completes the ComponentCatalog refactoring work for `v0.6`.  It merges the `ModuleCatalog` and `ComponentCatalog` types into a single type.

Follow up for #208.

### CODE REVIEW

File: src/Microsoft.ML.Core/CommandLine/CmdParser.cs [hunk 1]
```diff
@@ -290,8 +290,6 @@ public sealed class CmdParser
         private const int SpaceBeforeParam = 2;
         private readonly ErrorReporter _reporter;
         private readonly IHost _host;
-        // REVIEW: _catalog should be part of environment and can be get through _host.
```
Comment (TomFinley) [inline src/Microsoft.ML.Core/CommandLine/CmdParser.cs]: >REVIEW: _catalog should be part of environment and can be get through _host. [](start = 11, length = 76)

How about that. :)

File: src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs [hunk 2]
```diff
@@ -19,17 +19,6 @@ public static class TensorFlowUtils
...
-        // This method is needed for the Pipeline API, since ModuleCatalog does not load entry points that are located
-        // in assemblies that aren't directly used in the code. Users who want to use TensorFlow components will have to call
-        // TensorFlowUtils.Initialize() before creating the pipeline.
-        /// <summary>
-        /// Initialize the TensorFlow environment. Call this method before adding TensorFlow components to a learning pipeline.
-        /// </summary>
-        public static void Initialize()
-        {
-            ImageAnalytics.Initialize();
-        }
```
Comment (TomFinley) [inline src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs]: What is the new alternative to this, for the sake of the legacy API?

Comment (eerhardt) [reply src/Microsoft.ML.TensorFlow/TensorFlow/TensorflowUtils.cs]: We only have 2 of these empty `Initialize()` methods, so it wasn't really a holistic plan.
If someone is using the "new API" this isn't necessary at all.

If someone is using the legacy API, and needs to do this, they can follow the same approach outlined in the following issues:

* https://github.com/dotnet/machinelearning/issues/569#issuecomment-409343579
* https://github.com/dotnet/machinelearning/issues/559

Basically, they can just get the `typeof` some type in that assembly to ensure it is loaded up front.
